persistence/modeshape-persistence-file/src/main/java/org/modeshape/persistence/file/FileDb.java

```diff
@@ -102,13 +102,13 @@ public Document get( String key ) {
 
     @Override
     public List<SchematicEntry> load( Collection<String> keys ) {
+        final TransactionStore.TransactionMap<String, Document> txContent = transactionalContent(false);
+        final TransactionStore.TransactionMap<String, Document> actualContent = txContent != null ? txContent : persistedContent; 
         return keys.stream()
-                   .map(persistedContent::get)
+                   .map(actualContent::get)
                    .filter(Objects::nonNull)
-                   .map(document -> {
-                       SchematicEntry schematicEntry = () -> document;
-                       return schematicEntry;
-                   }).collect(Collectors.toList()); 
+                   .map(document -> (SchematicEntry) () -> document)
```

Comment:
It is possible to use `.map(SchematicEntry.class::class)` instead. Not sure if that's more readable, though.